Signal/src/call/CallService.swift

```diff
@@ -1507,13 +1543,26 @@ private class SignalCallData: NSObject {
                 assert(failedCall.callRecord != nil)
             }
 
+            // Ensure there is a call record.
+            let callType = (failedCall.direction == .incoming
+                ? RPRecentCallTypeIncomingIncomplete
+                : RPRecentCallTypeOutgoingIncomplete)
+            if let callRecord = failedCall.callRecord {
+                callRecord.updateCallType(callType)
```

Comment:
Won't this clobber the "missed" call record we just set up on line 1541? 

And in the non-missed case (`default:`, above) we're already asserting that a call record exists, so if this is just a belts and suspenders thing, can we only keep the else case.